apps/examples/testcase/le_tc/kernel/tc_libc_sched.c

```diff
@@ -82,6 +82,47 @@ static void tc_libc_sched_sched_get_priority_min(void)
 	TC_SUCCESS_RESULT();
 }
 
+/**
+ * @fn                   :tc_libc_sched_task_setcanceltype
+ * @brief                :This tc tests tc_libc_sched_task_setcanceltype()
+ * @Scenario             :The task_setcanceltype() function atomically both sets the calling
+ *                        task's cancelability type to the indicated type and returns the
+ *                        previous cancelability type at the location referenced by oldtype
+ *                        If successful pthread_setcanceltype() function shall return zero;
+ *                        otherwise, an error number shall be returned to indicate the error.
+ * @API'scovered         :task_setcanceltype
+ * @Preconditions        :none
+ * @Postconditions       :none
+ * @return               :void
+ */
+#ifndef CONFIG_CANCELLATION_POINTS
+static void tc_libc_sched_task_setcanceltype(void)
+{
+	int type;
+	int oldtype;
+	int ret_chk;
+	timer_t timer_id;
+	clockid_t clockid = 0; /*CLOCK_REALTIME*/
+	struct sigevent st_sigevent;
+	struct itimerspec st_timer_spec_val;
+
+	type = TASK_CANCEL_ASYNCHRONOUS;
+	oldtype = TASK_CANCEL_ASYNCHRONOUS;
+	ret_chk = task_setcanceltype(type, &oldtype);
+	TC_ASSERT_EQ("task_setcanceltype", ret_chk, OK);
+
+	ret_chk = timer_create(clockid, &st_sigevent, &timer_id);
```

Comment:
this API TC is blocks pthread TC in tc configuration.
To complete tc, i have used timer